pkg/chartmuseum/server.go

```diff
@@ -296,39 +297,47 @@ func (server *Server) addIndexObjectsAsync(index *repo.Index, objects []storage.
 	server.Logger.Debugw("Loading charts packages from storage (this could take awhile)",
 		"total", numObjects,
 	)
-	var err error
-	loaded := make([]*helm_repo.ChartVersion, numObjects)
-
-	var wg sync.WaitGroup
-	wg.Add(numObjects)
-	for idx, object := range objects {
-		go func(i int, o storage.Object) {
-			defer wg.Done()
-			if err == nil {
+
+	cvChan := make(chan *helm_repo.ChartVersion)
+	errChan := make(chan error)
+
+	// Provide a mechanism to short-circuit object downloads in case of error
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	for _, object := range objects {
+		go func(o storage.Object) {
+			select {
+			case <-ctx.Done():
+				return
+			default:
 				chartVersion, err := server.getObjectChartVersion(o, true)
 				if err != nil {
 					err = server.checkInvalidChartPackageError(o, err, "added")
+				}
+				if err != nil {
```

Comment:
This condition has already been met above, lines 317 and 318 can be removed

Reply:
With the current state of the code, this is unfortunately not true, as [checkInvalidChartPackageError](https://github.com/davidovich/chartmuseum/blob/96c5ef6426dad427292b606c84004f0b12c6a036/pkg/chartmuseum/server.go#L359) can nil out the error (in the case where the package is malformed in the store).

Reply:
ah i see, thanks for correcting.